iq.go

```diff
@@ -137,11 +31,13 @@ type IQPayload interface {
 	Namespace() string
 }
 
-// iqtype, from, to, id, lang string
-func NewIQ(a Attrs) IQ {
+// NewIQ iqtype, attrs
+func NewIQ(t IQType, a Attrs) IQ {
+	// TODO generate IQ ID if not set
```

Comment:
a little todo for you -> if you like.

every iq needs a id, that is importent

Reply:
Yes, at least ejabberd will not reply without and id.